lte/gateway/python/magma/mobilityd/tests/test_ip_allocator_dhcp.py

```diff
@@ -107,35 +109,44 @@ def run_dhcp_allocator_thread(
     ip_allocator_dhcp_fixture._monitor_thread.join()
 
 
+@patch("tempfile.NamedTemporaryFile")
 def test_allocate_ip_address(
+    mock_tempfile: MagicMock,
     ip_allocator_fixture: IPAllocatorDHCP,
     ip_desc_fixture: IPDesc,
     dhcp_desc_fixture: DHCPDescriptor,
 ) -> None:
+    mock_tempfile.return_value.__enter__.return_value.name = TMP_FILE
+    mock_tempfile.return_value.__exit__.side_effect = lambda *args: os.remove(TMP_FILE)
```

Comment:
Why do we need to mock tempfile instead of just using the real one?

Reply:
In this test we mock the subprocess run call and assert that it has been called with the expected arguments. Since the tempfile is an argument in the subprocess call and the tempfile is randomly generated, we need to mock the tempfile filename generation to get a predictable filename to compare in the asserts.